nifi-toolkit/nifi-toolkit-encrypt-config/src/main/groovy/org/apache/nifi/properties/ConfigEncryptionTool.groovy

```diff
@@ -1568,7 +1572,7 @@ class ConfigEncryptionTool {
 
                         // If the tool is not going to encrypt NiFiProperties and the existing file is already encrypted, encrypt and update the new sensitive props key
                         if (!tool.handlingNiFiProperties && existingNiFiPropertiesAreEncrypted) {
-                            AESSensitivePropertyProvider spp = new AESSensitivePropertyProvider(tool.keyHex)
+                            SensitivePropertyProvider spp = StandardSensitivePropertyProvider.fromKey(tool.keyHex)
```

Comment:
The field `tool.keyHex` probably needs to be re-examined as it is no longer _just_ the AES key. When the internal logic was written, it was designed with extensibility, but this tool was written as a helper script and not designed for the injection of the provider. This should be re-evaluated due to the introduction of the AWS KMS provider.

Reply:
I renamed `keyHex` to `keyOrKeyId` and `migrationKeyHex` to `migrationKeyOrKeyId`.  Similarly adjusted method parameter names, and various variable names in the class.

However, when I tried to rename either of the methods `getKey` and `getMigrationKey`, the tests hang.  Not sure where the behavior is coming from; even adding (not renaming) a similar method produces the hangs.

There is still plenty of internal hex-string handling, and I'll revisit those as time allows.  If we still can't resolve all of that, I'll open that secondary jira ticket.